addons/pos_viva_wallet/models/pos_payment_method.py

```diff
@@ -0,0 +1,207 @@
+    viva_wallet_client_secret = fields.Char(string="Client secret", copy=False)
+    viva_wallet_terminal_id = fields.Char(string="Terminal ID", help='[Terminal ID of the Viva Wallet terminal], for example: 16002169', copy=False)
+    viva_wallet_bearer_token = fields.Char(copy=False, groups='base.group_erp_manager', default='Bearer Token')
+    viva_wallet_webhook_verification_key = fields.Char()
+    viva_wallet_latest_response = fields.Json(copy=False, groups='base.group_erp_manager') # used to buffer the latest asynchronous notification from Adyen.
+    viva_wallet_test_mode = fields.Boolean(string="Test mode", help="Run transactions in the test environment.")
+
+    def _bearer_token(self):
+        if not self.env.user.has_group('point_of_sale.group_pos_user'):
+            raise AccessError(_("Do not have access to fetch token from Viva Wallet"))
+
+        data = {'grant_type': 'client_credentials'}
+        auth = requests.auth.HTTPBasicAuth(self.viva_wallet_client_id, self.viva_wallet_client_secret)
+        try:
+            resp = requests.post(self.viva_wallet_account_get_endpoint() + '/connect/token', auth=auth, data=data, timeout=TIMEOUT)
+        except requests.exceptions.RequestException:
+            _logger.exception("Failed to call viva_wallet_bearer_token endpoint")
+            raise UserError(_("There are some issues between us and Viva Wallet, try again later."))
+
+        if resp.status_code == 200:
+            self.viva_wallet_bearer_token = json.loads(resp.text).get('access_token')
+
+    def _is_write_forbidden(self, fields):
+        # Allow the modification of these fields even if a pos_session is open
+        whitelisted_fields = {'viva_wallet_bearer_token', 'viva_wallet_webhook_verification_key', 'viva_wallet_latest_response'}
+        return bool(fields - whitelisted_fields and self.open_session_ids)
+
+    def _get_payment_terminal_selection(self):
+        return super()._get_payment_terminal_selection() + [('viva_wallet', 'Viva Wallet')]
+
+    def _get_verification_key(self, endpoint, viva_wallet_merchant_id, viva_wallet_api_key):
+        # Get a key to configure the webhook.
+        # this key need to be the response when we receive a notifiaction
+        authentication = base64.b64encode(str(viva_wallet_merchant_id).encode() + b':' + str(viva_wallet_api_key).encode())
+        headers = {
+                'Authorization': 'Basic ' + authentication.decode()
+            }
+        if tools.config['test_enable']:
+            return str('viva_wallet_test')
+        try:
+            resp = requests.get(endpoint + '/api/messages/config/token', headers=headers, timeout=TIMEOUT)
+        except requests.exceptions.RequestException:
+            _logger.exception('Failed to call https://"%s"/api/messages/config/token endpoint', endpoint)
```

Comment:
```suggestion
            _logger.exception('Failed to call https://%s/api/messages/config/token endpoint', endpoint)
```

quotation mark is not needed